packages/svelte/src/internal/client/dom/blocks/svelte-element.js

```diff
@@ -120,6 +139,9 @@ export function element(node, get_tag, is_svg, render_fn, get_namespace, locatio
 					render_fn(element, child_anchor);
 				}
 
+				// we do this after calling `render_fn` so that child effects don't override `nodes`
+				/** @type {Effect} */ (current_effect).nodes = { start: element, end: element };
```

Comment:
which child effects could override nodes?

Reply:
Those created in `render_fn`:

```svelte
<svelte:element this={'div'}>
  <!-- this will set the `<svelte:element>` effect's nodes to be whatever the child's nodes are
  <Child />
</svelte:element>
```

Reply:
Because of `nodes.end = ..` inside `append`? Else you could rely on `assign_nodes` not setting the nodes if the object is already present

Reply:
Yeah. I guess if we call `assign_nodes` before `render_fn` then at least it avoids creating a `nodes` object that will immediately be overwritten — one less object for the binman. Will make that tweak